### PR TITLE
Streamline release process

### PR DESCRIPTION
**Pull request by cassava**
- Fix broken boost dependency resolution.
- Reduce Docker image size by extracting vtd conan package.

### CODE REVIEW

**Comment by cassava** [comment]
Alright this should be ready for review now.

File: Makefile.package [hunk 1]
```diff
@@ -22,12 +22,15 @@ DATE := $(shell date +"%Y%m%d")
...
 
 HAS_GIT := $(shell [ -d ${PROJECT_ROOT}/.git ] && echo "true")
+ifeq (${PROJECT_VERSION},unknown)
 ifeq (${HAS_GIT},true)
-	GIT_COMMIT_HASH  := $(shell git log -1 --format=%h)
-	GIT_COMMIT_DIRTY := $(shell git diff --quiet || echo "-dirty")
-	GIT_DESCRIBE     := ${GIT_COMMIT_HASH}${GIT_COMMIT_DIRTY}
+GIT_COMMIT_HASH  := $(shell git log -1 --format=%h)
+GIT_COMMIT_DIRTY := $(shell git diff --quiet || echo "-dirty")
+GIT_DESCRIBE     := $(shell git describe --dirty="-dirty" | sed -r "s/^v(.*)/\1/")
```
Comment (tobifalk) [inline Makefile.package:29]: `git describe` may not work very well in CI context when shallow clones are used, but there is still the VERSION file which can be used instead.

**Review by tobifalk — COMMENTED** [review]
As a minor point, I noticed that in [test_engine.bats](https://github.com/eclipse/cloe/blob/ben/fix-boost/tests/test_engine.bats#L14) the comments may be updated:
```
    #    cloe-launch -vv exec -P conantest.py -- usage -j > tests/test_engine_json_schema.json
    #    cloe-launch -vv exec -o:o with_vtd=True -P conantest.py -- usage -j > tests/test_engine_json_schema_with_vtd.json
```

**Comment by tobifalk** [comment]
At some point we should consider activating Conan's [full transitivity mode](https://docs.conan.io/en/latest/creating_packages/define_abi_compatibility.html#enabling-full-transitivity-in-package-id-modes), but this can be done in another PR.

**Review by tobifalk — APPROVED** [review]

File: Makefile.package (hunk 1 above)
Comment (cassava) [reply Makefile.package:29]: Precisely!

**Pull request merged**